packages/driver/src/cy/commands/actions/select.ts

```diff
@@ -64,18 +64,20 @@ export default (Commands, Cypress, cy) => {
       }
 
       // normalize valueOrText if its not an array
-      valueOrText = [].concat(valueOrText).map((v) => {
+      valueOrTextOrIndex = [].concat(valueOrTextOrIndex).map((v) => {
+        if (_.isNumber(v) && (!_.isInteger(v) || v < 0)) $errUtils.throwErrByPath('select.invalid_number', { args: { index: v } })
```

Comment:
I'd probably suggest keeping this multiline since it makes the line length a bit long for legibility. We don't have any hard rules or linting enforcement of line lengths however. 

```suggestion
        if (_.isNumber(v) && (!_.isInteger(v) || v < 0)) {
          $errUtils.throwErrByPath('select.invalid_number', { args: { index: v } })
        }
```